### PR TITLE
largest-series-product corner cases

### PR DESCRIPTION
**Pull request by petertseng**
As part of the work to have consistency between tracks as in https://github.com/exercism/todo/issues/104 and https://github.com/exercism/todo/issues/13. Comments are in individual commit messages.

### CODE REVIEW

File: largest-series-product/largest_series_product_tests.erl [hunk 1]
```diff
@@ -6,3 +6,15 @@ three_test() -> ?assert( 504 =:= largest_series_product:from_string("0123456789"
 five_test() -> ?assert( 15120 =:= largest_series_product:from_string("0123456789", 5) ).
 
 six_test() -> ?assert( 23520 =:= largest_series_product:from_string("73167176531330624919225119674426574742355349194934", 6) ).
+
+all_zeroes_test() -> ?assert( 0 =:= largest_series_product:from_string("0000", 2) ).
+
+all_contain_zeroes_test() -> ?assert( 0 =:= largest_series_product:from_string("99099", 3) ).
+
+too_long_test() -> ?assertException(error, function_clause, largest_series_product:from_string("123", 4) ).
```
Comment (petertseng) [inline largest-series-product/largest_series_product_tests.erl]: I'm not 100% sure on whether to test for failure via just letting it crash (as it is written now, the `function_clause`, or to make the code explicitly return `{error, not_enough_digits}` or something. Existing tests use both styles: https://github.com/exercism/xerlang/blob/master/triangle/triangle_tests.erl#L38 vs https://github.com/exercism/xerlang/blob/master/clock/clock_tests.erl#L6

Happy to do it either way.

**Comment by petertseng** [comment]
Also, in case you wonder, I don't mind if you merge "move to `exercises/` directory" before this one, I can deal with rebasing, should be easy enough.

**Comment by kytrinyx** [comment]
@tmcgilchrist @NobbZ would either of you have a moment to review this?

**Comment by NobbZ** [comment]
I've quite a lot on my schedule. Also my wife made me do other things
first, so there's much less time left as originally planned. If
@tmcgilchrist doesn't chime in here, I will take a look at it on Monday
morning, when I also look at the corresponding Haskell PR.

Katrina Owen <notifications@github.com> schrieb am Sa., 30. Jan. 2016 17:03:

**Comment by kytrinyx** [comment]
Understood, @NobbZ!

File: largest-series-product/largest_series_product_tests.erl (hunk 1 above)
Comment (petertseng) [inline largest-series-product/largest_series_product_tests.erl]: I settled on using `assertError` with `_` as the `TermPattern` after reading http://erlang.org/doc/apps/eunit/chapter.html#EUnit_macros - basically I don't care what kind of error it is, I just want it to be an error. I think that's the way to go, but let me know if you disagree of course.

Comment (tmcgilchrist) [inline largest-series-product/largest_series_product_tests.erl]: I'll have a look later today, I'm currently away from a computer.

**Comment by tmcgilchrist** [comment]
I'm good with this :8ball: 

I'll merge this and rebase/squash your PR @NobbZ (if you don't mind).

**Pull request merged**